src/main/kotlin/org/bilalkilic/kediatrhelper/services/CreateNewHandlerService.kt

```diff
@@ -0,0 +1,43 @@
+package org.bilalkilic.kediatrhelper.services
+
+import com.intellij.ide.fileTemplates.FileTemplateManager
+import com.intellij.ide.fileTemplates.FileTemplateUtil
+import com.intellij.openapi.components.Service
+import com.intellij.psi.PsiElement
+import org.bilalkilic.kediatrhelper.utils.TemplateFileConstants.CLASS
+import org.bilalkilic.kediatrhelper.utils.TemplateFileConstants.MESSAGE
+import org.bilalkilic.kediatrhelper.utils.TemplateFileConstants.RETURN
+import org.bilalkilic.kediatrhelper.utils.TemplateFileConstants.CLASS_NAME_SUFFIX
+import org.bilalkilic.kediatrhelper.utils.TemplateFileConstants.TEMPLATE_FILE_COMMAND_HANDLER
+import org.bilalkilic.kediatrhelper.utils.TemplateFileConstants.TEMPLATE_FILE_QUERY_HANDLER
+import org.bilalkilic.kediatrhelper.utils.getQueryReturnType
+import org.bilalkilic.kediatrhelper.utils.isQuery
+import org.jetbrains.kotlin.psi.KtFile
+
+@Service
+class CreateNewHandlerService {
+
+    fun create(element: PsiElement) {
+        val directory = element.containingFile.containingDirectory
+        val mainClass = (element.parent as KtFile).classes.firstOrNull() ?: return
```

Comment:
Again can change element to KtClass to avoid casting here

Reply:
done.